models/unit.js

```diff
@@ -47,4 +48,8 @@ class Unit{
     this.tile.remove(this);
     this.player.remove(this);
   }
+
+  resetRangeCounter(){
+    this.rangeCounter = UNIT_RANGE_COUNTER_RESET
```

Comment:
Will the rangeCounter not always be zero? I was wondering how come you used a const var?

Reply:
The constant may not be necessary, it's just to eliminate a 'magic' zero. The rangeCounter should initialise to zero but then increment on a successful moveTo() call

Reply:
:+1: